fdbclient/IdempotencyId.actor.cpp

```diff
@@ -202,4 +202,130 @@ void decodeIdempotencyKey(KeyRef key, Version& commitVersion, uint8_t& highOrder
 	reader >> commitVersion;
 	commitVersion = bigEndian64(commitVersion);
 	reader >> highOrderBatchIndex;
+}
+
+// Find the youngest or oldest idempotency id key in `range` (depending on `reverse`)
+// Write the timestamp to `*time` and the version to `*version` when non-null.
+ACTOR static Future<Optional<Key>> getBoundary(Reference<ReadYourWritesTransaction> tr,
+                                               KeyRange range,
+                                               Reverse reverse,
+                                               Version* version,
+                                               int64_t* time) {
+	RangeResult result = wait(tr->getRange(range, /*limit*/ 1, Snapshot::False, reverse));
+	if (!result.size()) {
+		return Optional<Key>();
+	}
+	if (version != nullptr) {
+		BinaryReader rd(result.front().key, Unversioned());
+		rd.readBytes(idempotencyIdKeys.begin.size());
+		rd >> *version;
+		*version = bigEndian64(*version);
+	}
+	if (time != nullptr) {
+		BinaryReader rd(result.front().value, IncludeVersion());
+		rd >> *time;
+	}
+	return result.front().key;
+}
+
+ACTOR Future<Void> cleanIdempotencyIds(Database db, double minAgeSeconds) {
```

Comment:
Either here or in the header file, we could clarify (like in the PR description) that this clears out any leaked ids (not used in the main path during commits).